demos/unity_states/unity/Assets/Scripts/BallStateBillboardViewer.cs

```diff
@@ -0,0 +1,29 @@
+using UnityEngine;
+using UnityEngine.UI;
+
+public class BallStateBillboardViewer : MonoBehaviour
+{
+    [SerializeField] private bool isPlatform;
+    [SerializeField] private BallStateDataSourcePlatform dataSourcePlatform;
+    [SerializeField] private BallStateDataSourceUnity dataSourceUnity;
+    [SerializeField] private Text typeView;
+    [SerializeField] private Text detailsView;
+
+    private IBallStateDataSource dataSource;
+
+    void Awake()
+    {
+        this.typeView.text = this.isPlatform ? "PLATFORM" : "UNITY";
+        this.dataSource = this.isPlatform ? this.dataSourcePlatform : this.dataSourceUnity;
+    }
+
+    void Update()
+    {
+        string pos = $"X: {this.dataSource.pos_x}\nY: {this.dataSource.pos_y}\nZ: {this.dataSource.pos_z}";
+        string euler = $"θX: {this.dataSource.euler_x}\nθY: {this.dataSource.euler_y}\nθZ: {this.dataSource.euler_z}";
+        string scale = $"X: {this.dataSource.scale_x}\nY: {this.dataSource.scale_y}\nZ: {this.dataSource.scale_z}";
+        string color = $"C: {this.dataSource.color}";
+        string detailsStr = $"{pos}\n{euler}\n{scale}\n{color}";
```

Comment:
```suggestion
        var pos = $"X: {this.dataSource.pos_x}\nY: {this.dataSource.pos_y}\nZ: {this.dataSource.pos_z}";
        var euler = $"θX: {this.dataSource.euler_x}\nθY: {this.dataSource.euler_y}\nθZ: {this.dataSource.euler_z}";
        var scale = $"X: {this.dataSource.scale_x}\nY: {this.dataSource.scale_y}\nZ: {this.dataSource.scale_z}";
        var color = $"C: {this.dataSource.color}";
        var detailsStr = $"{pos}\n{euler}\n{scale}\n{color}";
```

Reply:
I like having the verbosity of the variables - more of a code style thing but I can use `var` moving forward.

Reply:
sure - I don't think its important to commit to one style to rule them all here since we have so little c# code in the codebase. resolving.